p2p/net/swarm/swarm.go

```diff
@@ -748,6 +772,8 @@ func (s *Swarm) removeConn(c *Conn) {
 
 	cs := s.conns.m[p]
 
+	oldState := s.connectednessUnlocked(p)
+
 	if len(cs) == 1 {
 		delete(s.conns.m, p)
 		s.conns.Unlock()
```

Comment:
Autorelay tests are failing if no event `NotConnected` is emitted when `oldState` is `NotConnected` and `newState` is `NotConnected` (no state change).